nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/ProtectedNiFiProperties.java

```diff
@@ -510,22 +448,21 @@ private boolean isProviderAvailable(String protectionScheme) {
      */
     private String unprotectValue(String key, String retrievedValue) {
         // Checks if the key is sensitive and marked as protected
-        if (isPropertyProtected(key)) {
-            final String protectionScheme = getProperty(getProtectionKey(key));
+        if (!isPropertyProtected(key)) {
+            return retrievedValue;
+        }
+        final String protectionScheme = getProperty(getProtectionKey(key));
 
-            // No provider registered for this scheme, so just return the value
-            if (!isProviderAvailable(protectionScheme)) {
-                logger.warn("No provider available for {} so passing the protected {} value back", protectionScheme, key);
-                return retrievedValue;
-            }
+        if (protectionScheme.equals("unknown")) {
+            return retrievedValue;
+        }
 
-            try {
-                SensitivePropertyProvider sensitivePropertyProvider = getSensitivePropertyProvider(protectionScheme);
-                return sensitivePropertyProvider.unprotect(retrievedValue);
-            } catch (SensitivePropertyProtectionException e) {
-                throw new SensitivePropertyProtectionException("Error unprotecting value for " + key, e.getCause());
-            }
+        // try and make one to unprotect, and if that fails...
```

Comment:
Please add the guard check using `StandardSensitivePropertyProvider.handlesProtectionScheme()` or similar to log warning message and return protected value without attempting (and failing) to create a provider implementation for a protection scheme we can't handle.

Reply:
Added static method `StandardSensitivePropertyProvider#hasProviderFor(scheme)`, updated this call, and added logging at this call site.